hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -5169,6 +5169,10 @@ func (r *HostedClusterReconciler) reconcileKubevirtPlatformDefaultSettings(ctx c
 
 func (r *HostedClusterReconciler) reconcilePlatformDefaultSettings(ctx context.Context, hc *hyperv1.HostedCluster, createOrUpdate upsert.CreateOrUpdateFN, logger logr.Logger) error {
 	switch hc.Spec.Platform.Type {
+	case hyperv1.AzurePlatform:
+		if azureutil.IsAroHCP() {
```

Comment:
I'd prefer reduce the number of hardcoded IsAroHCP checks we have and rather integrate building blocks.
E.g. We can agree and document the API input will take precedent and if empty we will fallback to this environment variable.

Reply:
Resolving this conversation since this code has been removed since this comment.